test/e2e/nodepool_kv_advanced_multinet_test.go

```diff
@@ -0,0 +1,328 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	discoveryv1 "k8s.io/api/discovery/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/utils/pointer"
+	"k8s.io/utils/ptr"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	capkv1alpha1 "sigs.k8s.io/cluster-api-provider-kubevirt/api/v1alpha1"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	hcpmanifests "github.com/openshift/hypershift/control-plane-operator/hostedclusterconfigoperator/controllers/resources/manifests"
+	"github.com/openshift/hypershift/hypershift-operator/controllers/manifests"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+)
+
+type KubeVirtAdvancedMultinetTest struct {
+	infra        e2eutil.KubeVirtInfra
+	ifaceName    string
+	nodePoolName string
+}
+
+func NewKubeVirtAdvancedMultinetTest(ctx context.Context, mgmtClient crclient.Client, hc *hyperv1.HostedCluster) NodePoolTest {
+	return KubeVirtAdvancedMultinetTest{
+		infra:        e2eutil.NewKubeVirtInfra(ctx, mgmtClient, hc),
+		ifaceName:    "net1",
+		nodePoolName: hc.Name + "-" + "test-kv-advance-multinet",
+	}
+}
+
+func (k KubeVirtAdvancedMultinetTest) Setup(t *testing.T) {
+	if globalOpts.Platform != hyperv1.KubevirtPlatform {
+		t.Skip("test only supported on platform KubeVirt")
+	}
+
+	t.Log("Starting test KubeVirtAdvancedMultinetTest")
+}
+
+func (k KubeVirtAdvancedMultinetTest) Run(t *testing.T, nodePool hyperv1.NodePool, _ []corev1.Node) {
+	g := NewWithT(t)
+
+	np := &hyperv1.NodePool{}
+	g.Expect(k.infra.MGMTClient().Get(k.infra.Ctx(), util.ObjectKey(&nodePool), np)).Should(Succeed())
+	g.Expect(np.Spec.Platform).ToNot(BeNil())
```

Comment:
I think you'll find asserting over `cmp.Diff()` for structures quite a lot simpler and more readable for failures.

Reply:
Make sens I will use that.

Reply:
Looking at it again, we are looking at specific fields we don't care about the rest of them, using `cmp.Diff` means that we have to select/ignore the fields there, I don't thing is worth it on this case.

Reply:
What sub-sections of `np.Spec.Platform` are filled with _something_ and not asserted over?

Reply:
This is the NodePool kubevirt struct, from there we just one to check at the test a pair of fields not all of them, if we use cmp.Diff() then we have to care for the others.

```
type KubevirtNodePoolPlatform struct {                                          
    RootVolume *KubevirtRootVolume `json:"rootVolume"`                          
    Compute *KubevirtCompute `json:"compute"`                                   
    NetworkInterfaceMultiQueue *MultiQueueSetting `json:"networkInterfaceMultiqueue,omitempty"`
    AdditionalNetworks []KubevirtNetwork `json:"additionalNetworks,omitempty"`  
    AttachDefaultNetwork *bool `json:"attachDefaultNetwork,omitempty"`          
    NodeSelector map[string]string `json:"nodeSelector,omitempty"`              
```

Reply:
OK, up to you. I think it would be well worth the diff output. Did you try both of them out? Can you paste the output on failure for both cases? The output of `Expect()...` is pretty mangled and leaves the dev searching through the output to figure out what field is unset and why

Reply:
I am going to use gomega `BeComparableTo` is using google cmp https://onsi.github.io/gomega/#becomparabletoexpected-interface-options-cmpoption

Reply:
@stevekuznetsov so for example this is the comparation diff and as expected it show stuff that I don't care on the test 
```
Platform: v1beta1.NodePoolPlatform{
          			Type:     "KubeVirt",
          			AWS:      nil,
          			IBMCloud: nil,
          			Kubevirt: &v1beta1.KubevirtNodePoolPlatform{
        - 				RootVolume: &v1beta1.KubevirtRootVolume{
        - 					KubevirtVolume: v1beta1.KubevirtVolume{
        - 						Type:       "Persistent",
        - 						Persistent: &v1beta1.KubevirtPersistentVolume{Size: s"32Gi", VolumeMode: &"Filesystem"},
        - 					},
        - 				},
        + 				RootVolume:                 nil,
        - 				Compute:                    &v1beta1.KubevirtCompute{Memory: s"12Gi", Cores: &4, QosClass: &"Burstable"},
        + 				Compute:                    nil,
          				NetworkInterfaceMultiQueue: nil,
          				AdditionalNetworks:         {{Name: "default/e2e-test7mlkg"}},
          				... // 2 identical fields
          			},
          			Agent:   nil,
          			Azure:   nil,
          			PowerVS: nil,

```

Reply:
I am going to keep just the fiew checks on the struct, is not worth it start filtering fields out and the like, thanks for the cmp.Diff pointer though!.